kernelci/data/kernelci_api.py

```diff
@@ -72,6 +72,13 @@ def get_node(self, node_id):
         resp = self._get('/'.join(['node', node_id]))
         return json.loads(resp.text)
 
+    def get_node_by_commit_hash(self, commit_hash):
```

Comment:
This will actually return a list of nodes, with all the ones matching the git SHA1. So we probably should call it `get_nodes_by_commit_hash` and mention it in the docstring.